zfs/prometheus.go

```diff
@@ -34,6 +35,14 @@ func init() {
 		Name:      "destroy_duration",
 		Help:      "Duration it took to destroy a dataset",
 	}, []string{"dataset_type", "filesystem"})
+	prom.ZFSListUnmatchedUserSpecifiedRulesCount = prometheus.NewGaugeVec(prometheus.GaugeOpts{
+		Namespace: "zrepl",
+		Subsystem: "zfs",
+		Name:      "list_unmatched_user_specified_rules_count",
```

Comment:
Lets rename the metric and the variable name to

```
zrepl_zfs_list_unmatched_user_specified_datasets_count
```

That gets it in line with the name of the interface method.